### PR TITLE
add spec.versions

### PR DESCRIPTION
**Pull request by kalantar**
Signed-off-by: Michael Kalantar <kalantar@us.ibm.com>
Addresses https://github.com/iter8-tools/iter8/issues/963

### CODE REVIEW

**Comment by kalantar** [comment]
Lowered the coverage requirement to get passage ... we plan to make many changes. Some will temporarily lower test coverage. Will raise again later.

File: api/v2beta1/experiment_types.go [hunk 1]
```diff
@@ -56,6 +56,10 @@ type ExperimentSpec struct {
 	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
 
+	// Versions is list of version labels
```
Comment (sriumcp) [inline api/v2beta1/experiment_types.go:59]: Could I request renaming this as `VersionInfo`.
***
The principle is as follows: wherever we have a list of info items with each item associated with a specific version, we call the list as `VersionInfo`.
***
The length of `VersionInfo` lists need to be the same regardless of where they appear in the experiment.
***
`Spec.VersionInfo` specifies a list of version identifiers (strings).

**Review by sriumcp — CHANGES_REQUESTED** [review]
Please see comment on `versions` and `versionInfo`.

File: api/v2beta1/experiment_types.go (hunk 1 above)
Comment (kalantar) [reply api/v2beta1/experiment_types.go:59]: I would like to leave this as `Versions` for a short while. I have not yet removed the existing `VersionInfo`. I introduced `Versions` now to allow for removal of `Spec.Strategy.TestingPattern` (and computation of `Status.TestingPattern`). I am trying to remove and simplify more before taking the large step of removing `VersionInfo`.

Comment (sriumcp) [reply api/v2beta1/experiment_types.go:59]: Ok. This is fine as an interim step.

**Review by sriumcp — APPROVED** [review]
lgtm

**Pull request merged**